drivers/adc/adc_mcux_lpadc.c

```diff
@@ -57,13 +58,70 @@ struct mcux_lpadc_data {
 	lpadc_conv_command_config_t cmd_config[CONFIG_LPADC_CHANNEL_COUNT];
 };
 
+static int mcux_lpadc_acquisition_time_setup(const struct device *dev, uint16_t acq_time,
+					     lpadc_conv_command_config_t *cmd)
+{
+	const struct mcux_lpadc_config *config = dev->config;
+	uint32_t adc_freq_hz = 0;
+	uint32_t conversion_factor = 0;
+	uint32_t acquisition_time_value = ADC_ACQ_TIME_VALUE(acq_time);
+	uint8_t acquisition_time_unit = ADC_ACQ_TIME_UNIT(acq_time);
+
+	if (ADC_ACQ_TIME_DEFAULT == acquisition_time_value) {
+		return 0;
+	}
+
+	/* If the acquisition time is expressed in ADC ticks, then directly compare
+	 * the acquisition time with configuration items (3, 5, 7, etc. ADC ticks)
+	 * supported by the LPADC. The conversion factor is set to 1 (means do not need
+	 * to convert configuration items from ADC ticks to nanoseconds).
+	 * If the acquisition time is expressed in microseconds or nanoseconds, First
+	 * calculate the ADC cycle based on the ADC clock, then convert the configuration
+	 * items supported by LPADC into nanoseconds, and finally compare the acquisition
+	 * time with configuration items. The conversion factor is equal to the ADC cycle
+	 * (means convert configuration items from ADC ticks to nanoseconds).
+	 */
+	if (ADC_ACQ_TIME_TICKS == acquisition_time_unit) {
+		conversion_factor = 1;
+	} else {
+		if (clock_control_get_rate(config->clock_dev, config->clock_subsys, &adc_freq_hz)) {
+			LOG_ERR("Get clock rate failed");
+			return -EINVAL;
+		}
+
+		conversion_factor = (uint32_t)((float)1000000000.0 / adc_freq_hz);
```

Comment:
I know I asked this before, but now I still really don't get value of the intermediate conversion to float, since the max value here (1 billion, if I am reading the number correctly) is within the range of ints, and the fractional part will be truncated anyways by converting back to int, so there seems to be no increase in the range or precision by converting to float.... am I misunderstanding something